clerk.go

```diff
@@ -448,6 +448,32 @@ type ClientConfig struct {
 	BackendConfig
 }
 
+// Clock is an interface that can be used with the library instead
+// of the [time] package.
+// The interface is useful for testing time sensitive paths or
+// feeding the library with an authoritative source of time, like
+// an external time generator.
+type Clock interface {
+	Now() time.Time
+}
+
+// A default implementation of a Clock, keeping the real time by
+// using the [time] package directly.
+type defaultClock struct{}
+
+// Now returns the current time.
+func (c *defaultClock) Now() time.Time {
+	return time.Now()
```

Comment:
🔧 I believe we should use UTC by default
```suggestion
	return time.Now().UTC()
```

Reply:
Thanks. I'll update this in a follow up PR.

Reply:
https://github.com/clerk/clerk-sdk-go/pull/267